### ISSUE TITLE
Library is 64Kb, minified

### ISSUE THREAD

**Issue by NullVoxPopuli** [web-platform-dx/baseline-status#16]
Posting from: https://bsky.app/profile/nullvoxpopuli.com/post/3l4g6z3holy27

After gzip, this library (+ dependencies) are 23kb.

This is quite hefty.

BundlePhobia: https://bundlephobia.com/package/baseline-status@1.0.4

Some of it is lit -- but a lot of it is just this component.

some questions:
 - Does it need all of lit?
 - is this library side-effect free?
   - this would allow tree-shaking of unused consts/functions/etc
   - BundlePhobia says:
     Exports analysis is available only for packages that export ES Modules and are side-effect free.
     This package [...] isn't marked side-effect free.

**Comment by stof** [comment]
the package is not side-effect free as it registers a custom element (well, 2 of them actually, as there is an internal one)

**Comment by niutech** [comment]
Whooping [64KB of minified code](https://bundlephobia.com/package/baseline-status@1.0.8) for a small Baseline Status widget is over the roof. It should fit in <10KB of minified code.

**Comment by stof** [comment]
I think a big part of it is the string containing CSS code. A JS minifier won't be able to reduce the size of that string, which takes half of the size of the file.

**Comment by captainbrosset** [comment]
I agree this is a big problem that we need to solve. We shouldn't be imposing this performance cost to the hosting webpages.
(Out of scope for this issue, but we probably shouldn't be imposing JS at all either, and think about what a server-side rendered HTML version could be).

Looking at the code, the SVG browser icons seem to be responsible for a large portion of the total size of the source file. I'm estimating around 60% of it.
On top of this, the built version embeds Lit, which adds even more.

**Comment by captainbrosset** [comment]
Regarding the SVG, we need to think about simplifying them substantially. The size at which they get rendered by default is very small and doesn't warrant that much details which most users will not see.

If I save the SVG string for the Safari logo (which is the longest) as a SVG file, it comes out at 20KB!
Saving the rendered image (at its default size) as a PNG file results in a 3KB file only.

What are some techniques we can use for responsive images in the web component?

**Comment by niutech** [comment]
@captainbrosset As for no-JS widget, Astro Embed [`<BaselineStatus>`](https://astro-embed.netlify.app/components/baseline-status/) component is using Declarative Shadow DOM.

As for SVG optimization, there is an excellent tool [SVGOMG](https://jakearchibald.github.io/svgomg/).

**Comment by bramus** [comment]
Instead of trying to wrangle the SVGs, how well do base64-encoded small PNGs do?

**Comment by captainbrosset** [comment]
> Instead of trying to wrangle the SVGs, how well do base64-encoded small PNGs do?

Agreed that the SVGs are unnecessarily complex. At most we're only going to be showing these icons somewhere between 30px to 200px. At these sizes, raster images seem to be perform a lot better. In my previous comment, I tested AVIF which (even for the largest page zoom) produced a 3KB file for the Safari logo.

Someone should try to embed base64-encoded versions of these images and see what the result feels like.

**Comment by captainbrosset** [comment]
If we can live with the fact that, at max zoom level, the optimized SVG shows some incorrect details, then I really like the idea.
![image](https://github.com/user-attachments/assets/6851e011-c1c6-4000-b838-b8b83427873c)

**Comment by bramus** [comment]
What about not including any images in the package itself? The data is fetched from an external endpoint anyways, so maybe the API server could also serve the logos?

_(In that case, maybe also allow an option to set the baseURL for the logos, so that people can self-host the logos if they want to)_

**Comment by jsnkuhn** [comment]
I know that the SVGs you are talking about above are the browser icons but it occurred to me that the baseline icons can probably also be reduced in file size by using `stroke` to create them instead of `fill`. Here's a proof of concept that cuts the `baseline-newly-icon.svg` icon code in half:

https://codepen.io/jsnkuhn/pen/OPLZddP?editors=1100

**Comment by jsnkuhn** [comment]
What about the possibility of using a sprite sheet for the browser icons? 

Would reduce the number of HTTP requests the widget imposes on sites. Also usually the overall file size of a sprite sheet is smaller than the file sizes of the individual images added together.
